app/controllers/about/AboutPropertyCompleteController.scala

```diff
@@ -68,19 +68,26 @@ class AboutPropertyCompleteController @Inject() (
             for {
               updatedAnswers <- Future.fromTry(request.userAnswers.set(AboutPropertyCompletePage, value))
               _              <- sessionRepository.set(updatedAnswers)
-              _ <- journeyAnswersService
-                     .setStatus(
-                       ctx = JourneyContext(
-                         taxYear = taxYear,
-                         mtditid = request.user.mtditid,
-                         nino = request.user.nino,
-                         journeyName = "property-about"
-                       ),
-                       status = statusForPage(value),
-                       user = request.user
-                     )
-            } yield Redirect(
-              navigator.nextPage(AboutPropertyCompletePage, taxYear, mode, request.userAnswers, updatedAnswers)
+              status <- journeyAnswersService.setStatus(
```

Comment:
This looks fine, how about this? Extract user answer updating and status saving into separate private methods for clarity. (onSubmit method 32lines)

  ```
private def saveUserAnswersAndRedirect(value: YourFormType, taxYear: Int, mode: Mode)(implicit request: DataRequest[AnyContent]): Future[Result] = {
    for {
      updatedAnswers <- Future.fromTry(request.userAnswers.set(AboutPropertyCompletePage, value))
      _              <- sessionRepository.set(updatedAnswers)
      status         <- journeyAnswersService.setStatus(
                          JourneyContext(taxYear, request.user.mtditid, request.user.nino, "property-about"),
                          statusForPage(value),
                          request.user
                        )
    } yield status.fold(
      _ => internalServerError(taxYear, request.user),
      _ => Redirect(navigator.nextPage(AboutPropertyCompletePage, taxYear, mode, request.userAnswers, updatedAnswers))
    )
  }

  private def internalServerError(taxYear: Int, user: User): Result = {
    InternalServerError(
      s"Failed to save status for the about section, for the UK property journey for tax year: $taxYear, " +
      s"user with nino: ${user.nino} and mtditid: ${user.mtditid}"
    )
  }
```

Reply:
done